unittests/test_parameters.py

```diff
@@ -273,8 +273,26 @@ class Bar(Foo):
 
 
 def test_override_regular_attribute():
+    class Foo(rfm.RegressionTest):
+        p = 4
+        p = parameter([1, 2])
+
+    assert Foo.p == (1, 2,)
```

Comment:
This would not work. The parameters are only stored as instances of `TestParam` in the local parameter space, not in the class' final parameter space. This is because the `TestParam` class just stores the values of the parameters and their inheritance behaviour. Since the parameter inheritance is resolved when building the final class' parameter space, all this extra info in `TestParam` is no longer needed and the parameters are just stored in the parameter space dict as tuples with all their values.

Reply:
Right, now I remember that when I was playing with my other PR about the naming scheme.